src/ansi-c/c_typecheck_type.cpp

```diff
@@ -272,7 +274,7 @@ void c_typecheck_baset::typecheck_type(typet &type)
       else if(gcc_attr_mode == "__V4DF__") // deprecated vector of 4 doubles
         result=vector_typet(double_type(), from_integer(4, size_type()));
       else // give up, just use subtype
-        result=type.subtype();
+        result = to_type_with_subtype(type).subtype();
```

Comment:
Same here (with same caveat): UNREACHABLE?

Reply:
They aren't unreachable (the parser accepts any identifier). It might make sense to give a user-facing error message if we are confident that we've got them all.